alf/config_util.py

```diff
@@ -557,13 +561,21 @@ def _wrapper(*args, **kwargs):
                     unspecified_kw_args[name] = config.get_value()
                 config.set_used()
 
+        for key in guard_kwargs:
+            if key in kwargs and key in configs:
```

Comment:
We can just directly assert that any key in guard_kwargs shouldn't appear in kwargs or args?

Reply:
+1

Reply:
Done.